server/server.js

```diff
@@ -1,15 +1,158 @@
 const express = require("express");
 const app = express();
+require('dotenv').config()
 const port = process.env.PORT || 5000;
+// const AllVideos = require("")
+const cors = require("cors");
+const short = require('short-uuid');
+const translator = short("12345")
+app.use(express.json());
+app.use(cors());
+
+const { Pool } = require("pg");
+const db = new Pool({
+  user: process.env.DB_USER,
+  host: process.env.DB_HOST,
+  database: process.env.DB_NAME,
+  password: process.env.DB_PASSWORD,
+  port: process.env.DB_PORT,
+  // ssl: true
+});
+
+db.connect(function (err) {
+  if (err) {
+    return console.error('error: ' + err.message);
+  }
+  console.log('Connected to the postgresSQL server.');
+});
+
+
+
+const bodyParser = require("body-parser");
+app.use(bodyParser.json());
 
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+// let videos = AllVideos;
 
 // GET "/"
 app.get("/", (req, res) => {
   // Delete this line after you've confirmed your server is running
   res.send({ express: "Your Backend Service is Running" });
 });
+
+
+
+app.put("/videos/:id", async (req, res) => {
+  const videoId = req.params.id;
+  const { rating } = req.body;
+
+  try {
+    const updateQuery = "UPDATE videos SET rating = $1 WHERE id = $2";
+    const updateValues = [rating, videoId];
+    await db.query(updateQuery, updateValues);
+
+    res.status(200).send("sent");
+  } catch (error) {
+    console.error("Error updating rating:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+
+
+// app.put("/videos/:id", async (req, res) => {
+//   const videoID = req.params.id;
+//   try {
+//     const video = await db.query("UPDATE videos SET rating = rating + 1 WHERE id=$1", [videoID]);
+//     if (!video) {
+//       res.status(404).send("Video not found");
+//       console.log("There is no video with this ID!");
+//     } else {
+//       res.status(200).send("Video updated");
+//     }
+//   } catch (error) {
+//     console.error("Error updating video:", error);
+//     res.status(500).json({ error: "Internal server error", message: error.message });
+//   }
+// });
+
+
+app.post("/videos", (req, res) => {
+  let newVideoTitle = req.body.title
+  let newVideoURL = req.body.url
+  let newVideoRating = 0
+
+  if (!newVideoTitle || !newVideoURL || newVideoTitle === "" || newVideoURL === "") {
+    res.status(400).send("Fill in the missing fields")
```

Comment:
It can be useful in an error message to say _which_ fields were missing, so that it can be displayed to the user and they know what to fix :)